test/scheduler_test.rb

```diff
@@ -456,6 +456,59 @@
            /#{Resque::Scheduler.send(:internal_name)}: cage/
   end
 
+  test 'gracefully shuts down rufus-scheduler threads' do
+    class BeforeEnqueueJob
+      @queue = :quick
+
+      class << self
+        def before_enqueue_example(*)
+          job_has_been_enqueued_once = !set_before_enqueue_flag(true)
+          return false if job_has_been_enqueued_once
+
+          sleep 5 # sleep longer than test timeout
+          true
+        end
+
+        def enqueue_started?
+          Resque.redis.get('before_enqueue_job:enqueued') == "true"
+        end
+
+        def perform(*)
+        end
+
+        private
+
+        def set_before_enqueue_flag(value)
+          Resque.redis.setnx('before_enqueue_job:enqueued', value)
```

Comment:
What do you think about just using `set` instead of `setnx` here since that's easier to understand and then above you can clean it up a bit more by doing something like this:
```ruby
return false if enqueue_started?
enqueue_started!
...
```